mev-boost/server/service.go

```diff
@@ -348,90 +347,81 @@ func (m *BoostService) verifyConstraintProofs(responsePayload *BidWithInclusionP
 		return errMissingConstraint
 	}
 
-	if len(responsePayload.Proofs) != len(inclusionConstraints) {
+	if responsePayload.Proofs == nil {
+		log.Warn("[BOLT]: Nil proof")
+		return errNilProof
+	}
+
+	if len(responsePayload.Proofs.TransactionHashes) != len(inclusionConstraints) {
 		log.Warnf("[BOLT]: Proof verification failed - number of preconfirmations mismatch: proofs %d != constraints %d",
-			len(responsePayload.Proofs), len(inclusionConstraints))
+			len(responsePayload.Proofs.TransactionHashes), len(inclusionConstraints))
 		return errMismatchProofSize
 	}
 
-	// BOLT: verify preconfirmation inclusion proofs. If they don't match, we don't consider the bid to be valid.
-	if responsePayload.Proofs != nil {
-		// BOLT: remove unnecessary fields while logging
-		log.WithFields(logrus.Fields{})
+	// BOLT: remove unnecessary fields while logging
+	log.WithFields(logrus.Fields{})
 
-		log.WithField("len", len(responsePayload.Proofs)).Info("[BOLT]: Verifying constraint proofs")
+	log.WithField("len", len(responsePayload.Proofs.TransactionHashes)).Info("[BOLT]: Verifying merkle proofs")
```

Comment:
I'd remove the part where we delete the log fields, it was not a good choice. Also,
I'd change the log with just
`log.Infof("Verifying Merkle multiproof for %d transactions", len(...)`